modules/vaos/spec/requests/appointments_request_spec.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+RSpec.describe 'vaos appointments', type: :request do
+  include SchemaMatchers
+
+  let(:rsa_private) { OpenSSL::PKey::RSA.generate 4096 }
+
+  before do
+    Flipper.enable('va_online_scheduling')
+    sign_in_as(current_user)
+    allow_any_instance_of(VAOS::JWT).to receive(:cert).and_return(rsa_private)
+  end
+
+  context 'loa1 user with flipper enabled' do
+    let(:current_user) { build(:user, :loa1) }
+
+    it 'does not have access' do
+      get '/services/vaos/v0/appointments'
+      expect(response).to have_http_status(:forbidden)
+      expect(JSON.parse(response.body)['errors'].first['detail'])
+        .to eq('You do not have access to online scheduling')
+    end
+  end
+
+  context 'loa3 user' do
+    let(:current_user) { build(:user, :mhv) }
+    let(:start_date) { Time.now.utc.beginning_of_day + 7.hours }
+    let(:end_date) { Time.now.utc.beginning_of_day + 8.hours + 4.months }
+    let(:params) { { type: 'va', start_date: start_date, end_date: end_date } }
+
+    context 'with flipper disabled' do
+      it 'does not have access' do
+        Flipper.disable('va_online_scheduling')
+        get '/services/vaos/v0/appointments'
+        expect(response).to have_http_status(:forbidden)
+        expect(JSON.parse(response.body)['errors'].first['detail'])
+          .to eq('You do not have access to online scheduling')
+      end
+    end
+
+    context 'without a type' do
+      it 'has a parameter missing exception' do
+        get '/services/vaos/v0/appointments'
```

Comment:
I suggest being explicit with your missing query string parameter here.
That way in the future, if the `validate_params` lines get reordered or another exception case is added before the `type` check, this spec won't mysteriously start failing.
```suggestion
        get '/services/vaos/v0/appointments', params: params.except(:type)
```